AEPAssurance/Source/QuickConnect/QuickConnectService.swift

```diff
@@ -0,0 +1,216 @@
+    ///
+    func getDeviceStatus(clientID: String,
+                         orgID: String,
+                         completion: @escaping (Result<(sessionID: String, token: Int), AssuranceConnectionError>) -> Void) {
+
+        /// Bail out with failure, if we are unable to form a valid create device API request URL
+        let urlString = AssuranceConstants.QUICK_CONNECT_BASE_URL + "/status"
+        guard let requestURL = URL(string: urlString) else {
+            let error = AssuranceConnectionError.invalidURL(url: urlString)
+            Log.error(label: self.LOG_TAG, error.info.description)
+            completion(.failure(error))
+            return
+        }
+
+        let parameters = [AssuranceConstants.QuickConnect.KEY_ORGID: orgID, AssuranceConstants.QuickConnect.KEY_CLIENT_ID: clientID]
+
+        /// Bail out with failure, if we are unable to create the request body required for the API
+        guard let body = try? JSONSerialization.data(withJSONObject: parameters, options: .prettyPrinted) else {
+            let error = AssuranceConnectionError.invalidRequestBody
+            Log.error(label: self.LOG_TAG, error.info.description)
+            completion(.failure(error))
+            return
+        }
+
+        /// Create the request
+        let request = NetworkRequest(url: requestURL,
+                                     httpMethod: HttpMethod.post,
+                                     connectPayloadData: body,
+                                     httpHeaders: HEADERS,
+                                     connectTimeout: AssuranceConstants.Network.CONNECTION_TIMEOUT,
+                                     readTimeout: AssuranceConstants.Network.READ_TIMEOUT)
+
+        ServiceProvider.shared.networkService.connectAsync(networkRequest: request) { connection in
+            if !(connection.responseCode == HTTP_RESPONSE_CODES.HTTP_OK || connection.responseCode == 201) {
+                let error = AssuranceConnectionError.failedToGetDeviceStatus(statusCode: connection.responseCode ?? -1, responseMessage: connection.responseMessage ?? "Unknown error")
+                Log.error(label: self.LOG_TAG, error.info.description)
+                completion(.failure(error))
+                return
+            }
+
+            if let data = connection.data, let responseDict = try? JSONDecoder().decode([String: AnyCodable].self, from: data) {
+                let sessionID = responseDict["sessionUuid"]?.stringValue
+                let token = responseDict["token"]?.intValue
+
+                Log.debug(label: self.LOG_TAG, "Device status \(String(describing: responseDict))")
+                guard let sessionID = sessionID, let token = token else {
+                    if self.shouldRetryGetDeviceStatus {
+                        sleep(2)
```

Comment:
I think it might be good to cap the number of retrys to a large number (15 mins?) to prevent any scenario(more like a bug) that causes the device to poll indefinitely. 
If you agree, we can add it as a fast follow later given the beta timelines.

Reply:
I agree. I can create a ticket for this.